src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -697,6 +698,22 @@ namespace OpenMS
       return *this;
     }
 
+    /// get the file path to the first MS run
+    StringList getPrimaryMSRunPath() const
+    {
+      StringList ms_run_paths;
+      std::vector<SourceFile> sfs(this->getSourceFiles());
+      for (std::vector<SourceFile>::const_iterator it = sfs.begin(); it != sfs.end(); ++it)
+      {
+        String ms_run_location = it->getPathToFile() + "/" + it->getNameOfFile();
```

Comment:
I think this is correct for both windows and linux path (if a proper file URI is used in the first place).
e.g.:
file:///C:/Users/Benutzer/Desktop/Uniform%20Resource%20Identifier.html
file:///etc/fstab
In OpenMS we only support file based URI's so I guess it is fine.